newIDE/app/src/EventsSheet/InstructionEditor/TextEditorDialog.js

```diff
@@ -0,0 +1,172 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { t } from '@lingui/macro';
+
+import * as React from 'react';
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import HelpButton from '../../UI/HelpButton';
+import { Line, Column } from '../../UI/Grid';
+import ColorPicker from '../../UI/ColorField/ColorPicker';
+import MiniToolbar, { MiniToolbarText } from '../../UI/MiniToolbar';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+import { getSelectedEvents } from '../SelectionHandler';
+
+const gd = global.gd;
+
+const toolbarItemStyle = {
+  marginRight: 10,
+};
+
+const styles = {
+  sizeTextField: {
+    width: 90,
+    ...toolbarItemStyle,
+  },
+  toolbarItem: toolbarItemStyle,
+  checkbox: toolbarItemStyle,
+};
+
+type Props = {|
+  event: any,
+  open: boolean,
+  onClose: () => void,
+  onApply: () => void,
+|};
+
+type State = {|
+  searchText: string,
+  textValue: any,
+|};
+
+//TODO prendre en compte aussi les commentaires des groupes, vérifié le type de l'event getType()
+//Puis selon le cas retourné le text avec la methode approprié a l'event getComment() ou getName()
+//Voir comment recup l'event car ma façon de faire dans le state récupère ce text "\"
+export default class TextEditorDialog extends React.PureComponent<
+  Props,
+  State
+> {
+  recupereLecommentaire = () => {
+    return gd
+      .asCommentEvent(
+        getSelectedEvents(this.props.event).some(event => {
+          return event;
+        })
+      )
+      .getComment();
+  };
+
+  state = {
+    searchText: '',
+    textValue: this.recupereLecommentaire(),
```

Comment:
Basically, the idea of this component should be:

1. When creating the state, you read all the information that you can from the event passed as prop. You should do this according to the event type (see my other comment). If it's too much work, you can put this in a method as you already did ("recupereLecommentaire").
2. Then, the dialog is ONLY modifying the state (using classic React.setState) to change the text or the color.
3. If we click on cancel, call the onCancel prop and close the window (you close the window by setting the edited event to `null`). (see the idea?)
4. If we click on apply, call the onApply prop but before calling it, read the color and the text that are stroed in the state and apply them on the event passed as prop (so, according to the event type, you call setName or setComment, and setBackgroundColor, etc...). You probably can make a function inside TextEditorDialog for this.

This should limit the logic to do in EventsSheet to only dealing with the textEditedEvent - which is very clean.